lib/dialects/postgres/data-types.js

```diff
@@ -538,13 +451,7 @@ module.exports = BaseTypes => {
     }
   };
 
-  RANGE.prototype.toSql = function toSql() {
-    return BaseTypes.RANGE.types.postgres.subtypes[this._subtype.toLowerCase()];
-  };
-  RANGE.prototype.toCastType = function toCastType() {
-    return BaseTypes.RANGE.types.postgres.castTypes[this._subtype.toLowerCase()];
-  };
-
+  // TODO: Why are base types being manipulated??
```

Comment:
I think it kind of derives from the fact that basetypes.ARRAY is also the postgres.ARRAY type (since only postgres uses the array type) (yes, you could make the same argument about range and hstore, which are doing it the right way :D)

There really should be a specialized postgres.ARRAY type, so it works the same as it does for every other type

Reply:
Fixed that!